pallet/account-migration/src/lib.rs

```diff
@@ -16,174 +16,167 @@
 // You should have received a copy of the GNU General Public License
 // along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
 
+//! # Darwinia account migration pallet
+//!
+//! ## Overview
+//!
+//! Darwinia2 uses ECDSA as its signature algorithm instead of SR25519.
+//! These two algorithm are not compatible.
+//! Thus, an account migration is required.
+//!
+//! ## Technical detail
+//!
+//! Users must send an extrinsic themselves to migrate their account(s).
+//! This extrinsic should be unsigned, the reason is the same as `pallet-claims`.
+//! This extrinsic's payload must contain a signature to the new ECDSA address, signed by their
+//! origin SR25519 key.
+//!
+//! This pallet will store all the account data from Darwinia1 and Darwinia Parachain.
+//! This pallet's genesis will be write into the chain spec JSON directly.
+//! The data will be processed off-chain(ly).
+//! After the verification, simply perform a take & put operation.
+//!
+//! ```nocompile
+//! user -> send extrinsic -> verify -> put(storages, ECDSA, take(storages, SR25519))
+//! ```
+
 #![cfg_attr(not(feature = "std"), no_std)]
+#![deny(missing_docs)]
 
-#[cfg(test)]
-mod mock;
-#[cfg(test)]
-mod test;
+// TODO: update weight
 
-/// Type alias for currency AccountId.
-type AccountIdOf<R> = <R as frame_system::pallet::Config>::AccountId;
-/// Type alias for currency balance.
-type BalanceOf<T> =
-	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+#[cfg(test)]
+mod tests;
 
 // darwinia
-use dc_primitives::Balance;
+use dc_primitives::{AccountId as AccountId20, Balance, Index};
 // substrate
-use frame_support::traits::Currency;
-#[cfg(feature = "std")]
-use frame_support::traits::GenesisBuild;
-use sp_core::{
-	crypto::ByteArray,
-	sr25519::{Public, Signature},
-	H160,
-};
-use sp_io::hashing::blake2_256;
-use sp_runtime::{traits::Verify, AccountId32};
-use sp_std::vec::Vec;
+use frame_support::{log, pallet_prelude::*};
+use frame_system::{pallet_prelude::*, AccountInfo};
+use pallet_balances::AccountData;
+use sp_core::sr25519::{Public, Signature};
+use sp_io::hashing;
+use sp_runtime::traits::Verify;
+
+type AccountId32 = [u8; 32];
```

Comment:
Why not use `AccountId32` in sp_runtime, it meets our demand in this crate. No need to create another one at all.